src/en/flamecomics/src/eu/kanade/tachiyomi/extension/en/flamecomics/FlameComics.kt

```diff
@@ -4,62 +4,158 @@ import android.graphics.Bitmap
-
-                    Page(i, document.location(), imageUrls + COMPOSED_SUFFIX)
-                }
+        }
+        page--
+
+        var lastPage = page * 20 + 20
+        if (lastPage > manga.size) {
+            lastPage = manga.size
+        }
+        if (lastPage < 0) lastPage = 0
+        return MangasPage(manga.subList(page * 20, lastPage), lastPage < manga.size)
+    }
+
+    override fun searchMangaNextPageSelector(): String? = null
+    override fun popularMangaNextPageSelector(): String? = null
+    override fun latestUpdatesNextPageSelector(): String? = null
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request =
+        GET("$baseUrl/browse?search=$query&page=$page")
+
+    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+    override fun latestUpdatesRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun searchMangaSelector(): String = throw UnsupportedOperationException()
+    override fun popularMangaSelector(): String = "div[class^=SeriesCard_imageContainer] img"
+    override fun latestUpdatesSelector(): String =
+        "div[class^=SeriesCard_chapterImageContainer] img"
+
+    override fun popularMangaFromElement(element: Element): SManga = SManga.create().apply {
+        title = element.attr("abs:alt").substringAfter("$baseUrl/").replaceFirst("/", "")
+        val url = element.attr("abs:src")
+        thumbnail_url = getRealUrl(url).substringBefore("&")
+        setUrlWithoutDomain(getRealUrl(url).substringBefore("/thumb"))
+    }
+
+    override fun latestUpdatesFromElement(element: Element): SManga =
+        popularMangaFromElement(element)
+
+    override fun searchMangaFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        val seriesData = getJsonData<PageData>(document)?.props?.pageProps?.series
+
+        title = seriesData?.title + ""
+
+        thumbnail_url = "$cdn/series/${seriesData?.series_id}/${seriesData?.cover}"
+
+        description = seriesData?.description
+
+        when (seriesData?.status) {
+            "Ongoing" -> status = SManga.ONGOING
+            "Dropped" -> status = SManga.CANCELLED
+            "Hiatus" -> status = SManga.ON_HIATUS
+            "Completed" -> status = SManga.COMPLETED
+            else -> SManga.UNKNOWN
+        }
+
+        author = seriesData?.author
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val doc = response.asJsoup()
+        val jsonData = doc.getElementById("__NEXT_DATA__")?.data() ?: return listOf<SChapter>()
+        val pageData = json.decodeFromString<PageData>(jsonData)
+        return pageData.props.pageProps.chapters.map { chapter ->
+            SChapter.create().apply {
+                setUrlWithoutDomain("/series/${pageData.props.pageProps.series.series_id}/${chapter.token}")
```

Comment:
If `token` is regularly updated it will cause entries to be marked as unread, at least when it can't detect the chapter number.